### DISCUSSION TITLE
Meditech ADT & CWS messages out of sequence

### DISCUSSION THREAD

**Discussion by kevin-murrell** [nextgenhealthcare/connect#4550] [Q&A]
I wonder if anyone else has experienced this? We have two incoming HL7 feeds from Meditech: ADT and CWS But we have a big problem in that we are receiving the messages out of sequence! An example today is a pre-reg message ADT^A05 received before the CWS SIU^S12. The A05 has the new account number included, but at that stage we don't have an appointment record for the patient.

The messageId of the CWS message indicates it was generated internally by Meditech before the user pre-reg'd the appointment.

This is just one site, and I haven't seen this elsewhere. Is this unusual? I really can't think of a fix we could apply when the messages get to our Mirth instance.

Regards all, Kevin

**Comment by vijayamirtharajxavier** [comment]
If I understood your problem is right, then you can do check the whether patient mrn is available in the database before you process this message, eg: under preprocessor javascript, just extract the mrn & date from the msg and do select query with you db, if there is schedule for that particular mrn & date you can process that message for your inbound if not available then you can use your use cases whether can create appointment record first and post this inbound message. This way you can solve this issue.

**Comment by tonygermano** [comment]
Have you asked the site why they are sending you messages out of order? I'd definitely look to fix this upstream before trying to come up with a workaround.
